File: solutions/dart/space-age/1/lib/space_age.dart
const _periods = {
  'Mercury': 0.2408467,
  'Venus': 0.61519726,
  'Earth': 1.0,
  'Mars': 1.8808158,
  'Jupiter': 11.862615,
  'Saturn': 29.447498,
  'Uranus': 84.016846,
  'Neptune': 164.79132,
};

const _secondsInYear = 31557600;

class SpaceAge {
  num age({String planet = '', num seconds = 0}) {
    var planetSeconds = seconds / (_periods[planet] ?? 0);
    var ageInCentiYears = planetSeconds / _secondsInYear * 100;
    return ageInCentiYears.roundToDouble() / 100;
  }
}

File: solutions/haskell/space-age/1/src/SpaceAge.hs
module SpaceAge (Planet(..), ageOn) where

data Planet = Mercury
            | Venus
            | Earth
            | Mars
            | Jupiter
            | Saturn
            | Uranus
            | Neptune

ageOn :: Planet -> Float -> Float
ageOn planet seconds = seconds / (yearOnPlanet planet)
  where earthYear = 31557600
        yearOnPlanet Mercury = 0.2408467 * earthYear
        yearOnPlanet Venus = 0.61519726 * earthYear
        yearOnPlanet Earth = earthYear
        yearOnPlanet Mars = 1.8808158 * earthYear
        yearOnPlanet Jupiter = 11.862615 * earthYear
        yearOnPlanet Saturn = 29.447498 * earthYear
        yearOnPlanet Uranus = 84.016846 * earthYear
        yearOnPlanet Neptune = 164.79132 * earthYear

        
        
